addons/stock_account/models/product.py

```diff
@@ -376,103 +377,156 @@ def _run_fifo_vacuum(self, company=None):
+        self._create_fifo_vacuum_anglo_saxon_expense_entries(list(zip(vacuum_svls, real_time_svls_to_vacuum)))
 
+    @api.model
+    def _create_fifo_vacuum_anglo_saxon_expense_entries(self, vacuum_pairs):
+        """ Batch version of _create_fifo_vacuum_anglo_saxon_expense_entry
+        """
+        AccountMove = self.env['account.move'].sudo()
+        account_move_vals = []
+        vacuum_pairs_to_reconcile = []
+        svls_accounts = {}
+        for vacuum_svl, svl_to_vacuum in vacuum_pairs:
+            if not vacuum_svl.company_id.anglo_saxon_accounting or not svl_to_vacuum.stock_move_id._is_out():
+                continue
+            account_move_lines = svl_to_vacuum.account_move_id.line_ids
+            # Find related customer invoice where product is delivered while you don't have units in stock anymore
+            reconciled_line_ids = list(set(account_move_lines._reconciled_lines()) - set(account_move_lines.ids))
+            account_move = AccountMove.search([('line_ids', 'in', reconciled_line_ids)], limit=1)
+            # If delivered quantity is not invoiced then no need to create this entry
+            if not account_move:
+                continue
+            accounts = svl_to_vacuum.product_id.product_tmpl_id.get_product_accounts(fiscal_pos=account_move.fiscal_position_id)
+            if not accounts.get('stock_output') or not accounts.get('expense'):
+                continue
+            svls_accounts[svl_to_vacuum.id] = accounts
+            description = "Expenses %s" % (vacuum_svl.description)
+            move_lines = vacuum_svl.stock_move_id._prepare_account_move_line(
+            vacuum_svl.quantity, vacuum_svl.value * -1,
+            accounts['stock_output'].id, accounts['expense'].id,
+            description)
+            account_move_vals.append({
+                'journal_id': accounts['stock_journal'].id,
+                'line_ids': move_lines,
+                'date': self._context.get('force_period_date', fields.Date.context_today(self)),
+                'ref': description,
+                'stock_move_id': vacuum_svl.stock_move_id.id,
+                'move_type': 'entry',
+            })
+            vacuum_pairs_to_reconcile.append((vacuum_svl, svl_to_vacuum))
+        new_account_moves = AccountMove.create(account_move_vals)
+        new_account_moves._post()
+        for new_account_move, (vacuum_svl, svl_to_vacuum) in zip(new_account_moves, vacuum_pairs_to_reconcile):
+            account = svls_accounts[svl_to_vacuum.id]['stock_output']
+            to_reconcile_account_move_lines = vacuum_svl.account_move_id.line_ids.filtered(lambda l: not l.reconciled and l.account_id == account and l.account_id.reconcile)
+            to_reconcile_account_move_lines += new_account_move.line_ids.filtered(lambda l: not l.reconciled and l.account_id == account and l.account_id.reconcile)
+            to_reconcile_account_move_lines.reconcile()
+
+    #TODO remove in master
```

Comment:
I guess you made the `TODO` for the potential overides but you don't call it any more anyway :s

Reply:
@Whenrow Yeah true. I was thinking just in case some customization explicitly called `_create_fifo_vacuum_anglo_saxon_expense_entry` but really it was more to be compliant with the stable policy.
Do you think I should simply remove the singleton version of the function ?

Reply:
It's ok to me to remove it as we don't call it anymore

Reply:
@Whenrow After discussing with STS from the SH team he is not really keen on removing private methods in stable, even though it's not explicitly called from the standard code. His reasons are custom code on Odoo.SH and possibly preventing some on-premise servers to restart when they call the removed method on startup (I know it's highly unlikely for this specific case). Are you really against leaving that there and only removing it in master ?

Reply:
@Aurelienvd Absolutely not. If STS has some reserves against it, leave it :)

Reply:
Ok, I added it back :+1: